lib/requestIterator.js

```diff
@@ -3,27 +3,33 @@
 const hyperid = require('hyperid')(true)
 const inherits = require('util').inherits
 const requestBuilder = require('./httpRequestBuilder')
+const clone = require('clone')
 
 function RequestIterator (opts) {
   if (!(this instanceof RequestIterator)) {
     return new RequestIterator(opts)
   }
 
   this.resetted = false
-  this.context = {}
+  this.initialContext = opts.initialContext || {}
+  this.resetContext()
   this.reqDefaults = opts
   this.requestBuilder = requestBuilder(opts)
   this.setRequests(opts.requests)
 }
 
 inherits(RequestIterator, Object)
 
+RequestIterator.prototype.resetContext = function () {
+  this.context = clone(this.initialContext)
```

Comment:
please use the `rdfc` module instead.

Reply:
Aye, done. Also replaced another pre-existing use of `clone`.

Reply:
Turns out, `rfdc` doesn't support copying `Buffer`s. I'll take that separately. Reverted the last commit.